### PR TITLE
gh-107801: Improve the accuracy of io.IOBase.seek docs

### PR DESCRIPTION
**Pull request by erlend-aasland**
- Add param docstrings
- Link to os.SEEK_* constants
- Mention the return value in the initial paragraph

* Issue: gh-107801

----
:books: Documentation preview :books:: https://cpython-previews--108268.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/io.rst [hunk 1]
```diff
@@ -403,21 +403,20 @@ I/O Base Classes
...
+      interpreted relative to the position indicated by *whence*,
+      and return the new absolute position.
+      Values for *whence* are:
 
-      * :data:`!SEEK_SET` or ``0`` -- start of the stream (the default);
+      * :data:`os.SEEK_SET` or ``0`` -- start of the stream (the default);
         *offset* should be zero or positive
-      * :data:`!SEEK_CUR` or ``1`` -- current stream position; *offset* may
+      * :data:`os.SEEK_CUR` or ``1`` -- current stream position; *offset* may
         be negative
```
Comment (AA-Turner) [inline Doc/library/io.rst]:
```suggestion
      * :data:`os.SEEK_CUR` or ``1`` -- current stream position;
        *offset* may be negative
```

File: Doc/library/io.rst [hunk 2]
```diff
@@ -403,21 +403,20 @@ I/O Base Classes
...
-      * :data:`!SEEK_END` or ``2`` -- end of the stream; *offset* is usually
+      * :data:`os.SEEK_END` or ``2`` -- end of the stream; *offset* is usually
         negative
```
Comment (AA-Turner) [inline Doc/library/io.rst]:
```suggestion
      * :data:`os.SEEK_END` or ``2`` -- end of the stream;
      *offset* is usually negative
```

**Review by AA-Turner — APPROVED** [review]
Thanks!

A

File: Doc/library/io.rst [hunk 3]
```diff
@@ -403,20 +403,19 @@ I/O Base Classes
...
-      * :data:`!SEEK_CUR` or ``1`` -- current stream position; *offset* may
-        be negative
-      * :data:`!SEEK_END` or ``2`` -- end of the stream; *offset* is usually
-        negative
-
-      Return the new absolute position.
+      * :data:`os.SEEK_CUR` or ``1`` -- current stream position;
+        *offset* may be negative
+      * :data:`os.SEEK_END` or ``2`` -- end of the stream;
+      *offset* is usually negative
```
Comment (erlend-aasland) [inline Doc/library/io.rst]:
```suggestion
        *offset* is usually negative
```

**Pull request merged**